src/uct/ugni/rdma/ugni_rdma_ep.c

```diff
@@ -302,6 +295,14 @@ ucs_status_t uct_ugni_ep_atomic_cswap64(uct_ep_h tl_ep, uint64_t compare, uint64
     return uct_ugni_post_fma(iface, ep, &fma->super, UCS_INPROGRESS);
 }
 
+#if GNI_VERSION < GNI_VERSION_CHECK(0x00, 0x5c, 0x0101)
```

Comment:
I doubt you have access to any gni-headers/ugni software that doesn't support Aries, unless ORNL has an ancient system.  You can always check for the presence of ```GNI_DEVICE_ARIES``` being defined in gni_pub.h  At runtime you can use ```GNI_GetDeviceType``` to determine whether the network is Aries or Gemini.
If you link dynamically against libugni you should be able to run your executable on either amd/gem or intel/aries assuming you don't compile for specific x86_64 cpu type.